### ISSUE TITLE
composer require fails with v2

### ISSUE THREAD

**Issue by sylouuu** [stefanzweifel/laravel-stats#166]
Hello,

I've updated my app to L6 and it works well. PHP 7.3.

But I can't install laravel-stats:

```
→ composer require "wnx/laravel-stats" --dev
Using version ^2.0 for wnx/laravel-stats
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Installation request for symfony/console (locked at v4.4.4) -> satisfiable by symfony/console[v4.4.4].
    - Conclusion: don't install phploc/phploc 4.0.0
    - Installation request for wnx/laravel-stats ^2.0 -> satisfiable by wnx/laravel-stats[v2.0.0].
    - Conclusion: don't install sebastian/version 3.0.0|install phploc/phploc 4.0.0
    - Conclusion: remove sebastian/version 3.0.0|install phploc/phploc 4.0.0
    - wnx/laravel-stats v2.0.0 requires phploc/phploc ~4.0|~5.0 -> satisfiable by phploc/phploc[4.0.0, 4.0.1, 5.0.0].
    - phploc/phploc 4.0.1 requires sebastian/version ^2.0 -> satisfiable by sebastian/version[2.0.0, 2.0.1].
    - phploc/phploc 5.0.0 requires sebastian/version ^2.0 -> satisfiable by sebastian/version[2.0.0, 2.0.1].
    - Can only install one of: sebastian/version[2.0.1, 3.0.0].
    - Can only install one of: sebastian/version[2.0.0, 3.0.0].
    - Installation request for sebastian/version (locked at 3.0.0) -> satisfiable by sebastian/version[3.0.0].


Installation failed, reverting ./composer.json to its original content.
```

Do you have an idea?

Bests

**Comment by stefanzweifel** [comment]
Hi 👋 

The output of composer require says:
> Conclusion: don't install phploc/phploc 4.0.0

However, laravel-stats requires `phploc/phploc:~5.0`.

https://github.com/stefanzweifel/laravel-stats/blob/ac14fbade26b210ab2398336a79cd4fddef9beeb/composer.json#L19

Also `sebastian/version` is mentioned multiple times. Did you require any of these dependencies in your projects main `composer.json`? 

Maybe you have to upgrade `phpunit/phpunit`? Or some other dependencies? Maybe you can run `composer outdated` and update some of your dependencies?

----

According to our test suite, the package can be installed with Laravel 6 on php 7.3

- https://github.com/stefanzweifel/laravel-stats/runs/446872047?check_suite_focus=true#step:4:10
- https://github.com/stefanzweifel/laravel-stats/runs/446872037?check_suite_focus=true#step:4:10

**Comment by sylouuu** [comment]
Thanks for your answer. The thing is I don't require these packages, they are all deep.

Everything is up to date, an extract of my composer.json:

```json
{
    "require": {
        "php": ">=7.2",
        "adaojunior/passport-social-grant": "4.0.*",
        "aws/aws-sdk-php": "3.133.*",
        "barryvdh/laravel-snappy": "0.4.*",
        "caouecs/laravel-lang": "4.0.*",
        "cloudflare/sdk": "1.1.*",
        "doctrine/dbal": "2.10.*",
        "egulias/email-validator": "2.1.*",
        "fruitcake/laravel-cors": "1.0.*",
        "fzaninotto/faker": "1.9.*",
        "laravel/framework": "6.15.*",
        "laravel/passport": "8.4.*",
        "laravel/socialite": "4.3.*",
        "league/csv": "9.5.*",
        "league/flysystem-aws-s3-v3": "1.0.*",
        "league/fractal": "0.19.*",
        "mlocati/ip-lib": "1.9.*",
        "phpoffice/phpspreadsheet": "1.10.*",
        "predis/predis": "1.1.*",
        "sendinblue/api-v3-sdk": "6.1.*",
        "sentry/sentry-laravel": "1.6.*",
        "spatie/laravel-backup": "6.7.*",
        "spatie/laravel-image-optimizer": "1.5.*",
        "spatie/laravel-medialibrary": "7.18.*",
        "stripe/stripe-php": "7.25.*",
        "therobfonz/laravel-mandrill-driver": "1.0.*",
        "zanysoft/laravel-zip": "1.0.*"
    },
    "require-dev": {
        "barryvdh/laravel-ide-helper": "2.6.*",
        "beyondcode/laravel-dump-server": "1.3.*",
        "brianium/paratest": "4.0.*",
        "fabpot/goutte": "4.0.*",
        "lanin/laravel-api-debugger": "3.5.*",
        "laravel/tinker": "2.2.*",
        "mockery/mockery": "^1.3",
        "phpunit/phpunit": "9.0.*",
        "riimu/kit-phpencoder": "2.4.*"
    }
}
```

I'm a bit stuck :(

**Comment by stefanzweifel** [comment]
Thanks for providing your `composer.json`! This makes debugging such issues so much easier.

By using `composer why phploc/phploc` I could figure out, that the problem lies in the incompatibility of `sebastian/version`, `phpunit/phpunit:9.*` and `phploc/phploc`.

There's already an open issue in the phploc repository: https://github.com/sebastianbergmann/phploc/issues/204. However, the proposed solution of *just* using the phar, will not work for us 😬.

I will have to do some further testing/investigation, on how we could resolve this issue quickly, as I assume a lot of people will upgrade their projects to use phpunit v9 soon.

---

In the meantime I've released `v2.0.1` which adds support for phpunit v9.

**Comment by sylouuu** [comment]
```
→ composer why phploc/phploc

  [InvalidArgumentException]
  Could not find package "phploc/phploc" in your project
````

Thanks for doing investigations! :)

**Comment by stefanzweifel** [comment]
Hello again,

The upstream issue in `phploc/phploc` has been closed and a new major version of `phploc` has been released.
I've upped the version constraints in this package in #167 and released a new patch version [v2.0.2](https://github.com/stefanzweifel/laravel-stats/releases/tag/v2.0.2).

However, I couldn't reproduce your error anymore today on my machine (even before releasing v2.0.2).

I'm closing this issue for now. If you still can't install the package with the same error message, please just reopen this issue.

**Comment by sylouuu** [comment]
Great job @stefanzweifel it's OK now!

Have a nice day!
